website/docs/reference/widgets/table/inline-editing.md

```diff
@@ -1,197 +1,173 @@
+[Regular expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions) (regex) are patterns that describe valid user input. For example, the pattern `[a-zA-Z]` describes a string that contains only letters of the English alphabet. Using regular expressions in the **Regex** property of columns, you can set specific constraints on the input you expect from the user.
 
-For example, add a regular expression for entering a name. The name can contain only alphabets and spaces between the first and last name.
+When you add a regular expression to a column, all user input in that column is compared to the pattern; it's considered to be valid when it matches the expression, or invalid when it doesn't match. When input is invalid, the cell displays its **Error message** to the user in a tool-tip.
+
+For example, add a regular expression for entering a name. The name can contain only alphabetical characters and spaces between the first and last name:
 
 ```
 /^[a-z -]+$/i
 ```
 
-If you enter a value other than an alphabet or space (number of special characters), the widget shows an error message "invalid input."
-
-Similarly, you can use `s.`to only display words that start with the letter "**s**".
+If you enter a value that contains something other than alphabetical or space characters, the widget shows an error message "Invalid input."
 
-### Valid
+#### Valid
 
-Valid property shows the validity of the cell. When the expression evaluates to `false`, the input is considered invalid and the widget shows an Error Message. The following variables are available for binding:
+The **Valid** property is an expression that determines whether the cell input is acceptable. When this expression evaluates to `false`, the input is considered invalid and the widget shows its **Error Message**. The following variables are available for binding:
 
-1. `currentRow` - gives access to the row values of the editable cell.
+1. `currentRow` - accesses the values of the whole row that contains the editable cell.
 2. `currentIndex` - index of the current editable row.
-3. `editedValue` - the newly entered value on the editable input.
+3. `editedValue` - the newly entered value of the editable cell.
 
-Let's take an example to understand how the valid property works. If you want the updated value to be "John", so in the valid property section, add:
+For example, imagine you want the updated value to be `John`. In the **Valid** property field, add:
 
 ```
 {{editedValue == "John"}}
 ```
 
-If a word other than "John" is added to the cell, an error will be displayed. Similarly, the above-mentioned binding variables can be used to obtain values and row indexes.&#x20;
-
+If a value other than "John" is added to the cell, an error is displayed. Similarly, the previously mentioned binding variables can be used to obtain other values and row indexes.&#x20;
```

Comment:
Could we add a cross-reference to the binding variables section in the doc? It would help users to look at them if they need to.

On second thought, maybe you could add a cross-reference to the error message section below. It already has a screenshot there.